compiler/ballerina-backend-jvm-old/src/main/ballerina/src/compiler_backend_jvm/jvm_type_gen.bal

```diff
@@ -643,7 +643,7 @@ function addObjectAttachedFunctions(jvm:MethodVisitor mv, bir:BAttachedFunction?
             mv.visitVarInsn(ASTORE, attachedFunctionVarIndex);
 
             // if this initializer function, set it to the object type
-            if (stringutils:contains(attachedFunc.name.value, "__init")) {
+            if (stringutils:contains(attachedFunc.name.value, "$__init$")) {
```

Comment:
Shouldn't we check for the exact match of the $__init$?

Reply:
+1